go/dhcp/main.go

```diff
@@ -457,9 +489,24 @@ func (h *Interface) ServeDHCP(ctx context.Context, p dhcp.Packet, msgType dhcp.M
 				}
 			}
 
+			// Add options on the fly from pffilter
+			for key, value := range info.(map[string]interface{}) {
+				if key == "reject" {
+					log.LoggerWContext(ctx).Info("DHCPNAK on to " + clientMac)
+					answer.D = dhcp.ReplyPacket(p, dhcp.NAK, handler.ip.To4(), nil, 0, nil)
+					return answer
+				}
+				if s, ok := value.(string); ok {
+					var opcode dhcp.OptionCode
+					intvalue, _ := strconv.Atoi(key)
+					opcode = dhcp.OptionCode(intvalue)
+					GlobalOptions[opcode] = Tlv.Tlvlist[int(opcode)].Transform.Encode(s)
+				}
+			}
+
 			// Add device (mac) options on the fly
-			x, err = decodeOptions(p.CHAddr().String())
-			if err {
+			x, erreur = decodeOptions(p.CHAddr().String())
```

Comment:
Wi wile catch ze erreur my frende

Or perhaps we could catch err like everywhere else or something that is in english if it needs to have a different name